builders/build/substrate-api/py-substrate-interface.md

```diff
@@ -0,0 +1,311 @@
+
+# Query basic account information
+account_info = ws_provider.query(
+    module='System',
+    storage_function='Account',
+    params=['0x578002f699722394afc52169069a1FfC98DA36f1']
+)
+# Log the account nonce
+print(account_info.value['nonce'])
+# Log the account free balance
+print(account_info.value['data']['free'])
+
+# Query candidate pool information from Moonbeam's Parachain Staking module
+candidate_pool_info = ws_provider.query(
+    module='ParachainStaking',
+    storage_function='CandidatePool',
+    params=[]
+)
+print(candidate_pool_info)
+```
+
+## Signing and Transactions {: #signing-and-transactions }
+
+### Creating a Keypairs {: #creating-a-keypair }
+
+The Keypair object in Python Substrate Interface is used in the signing of any data, whether it's a transfer, a message, or a contract interaction.  
+
+You can create a Keypair instance from the shortform private key or from the mnemonic. For Moonbeam networks, you also need to specify the KeypairType to be `KeypairType.ECDSA`.
+
+#### Signing Data {: #signing-data }
+
+You can sign any arbitrary data using the keypair through the [`sign`](https://polkascan.github.io/py-substrate-interface/#substrateinterface.Keypair.sign){target=_blank} method.
+
+### Forming and Sending a Transaction {: #forming-and-sending-a-transaction }
+
+The [`compose_call`](https://polkascan.github.io/py-substrate-interface/#substrateinterface.SubstrateInterface.compose_call){target=_blank} method can be used to compose a call payload which can be used as an unsigned extrinsic or a proposal.
+
+Then the payload can be signed using a Keypair through the [`create_signed_extrinsic`](https://polkascan.github.io/py-substrate-interface/#substrateinterface.SubstrateInterface.create_signed_extrinsic){target=_blank} method. 
+
+The signed extrinsic can then be submitted using the [`submit_extrinsic`](https://polkascan.github.io/py-substrate-interface/#substrateinterface.SubstrateInterface.submit_extrinsic){target=_blank} method. 
+
+This method will also return an `ExtrinsicReceipt` object which contains information about the on-chain execution of the extrinsic. You can set the `wait_for_inclusion` to `True` when submitting the extrinsic if you need to examine the receipt object, to wait until the extrinsic is successfully included into the block. 
+
+The following sample code will show a complete example. 
+
+```python
+
+# Import
+from substrateinterface import SubstrateInterface
+
+# Construct the API provider
+ws_provider = SubstrateInterface(
+    url="{{ networks.moonbase.wss_url }}",
+) 
+
+# Define the Ethereum TxHash to Check Finality
+privatekey = bytes.fromhex("enter-shortform-private-key")
+mnemonic = 'enter-account-mnemonic'
+
+# Generate the keypair
+keypair = Keypair.create_from_private_key(privatekey, crypto_type=KeypairType.ECDSA)
+#keypair = Keypair.create_from_mnemonic(mnemonic, crypto_type=KeypairType.ECDSA)
+
+# Sign a message
+signature = keypair.sign("TestMoonbeam")
+print(signature)
+
```

Comment:
ah ok I see an example is here, but I think there should be snippets in the previous sections